service-base/src/main/java/org/eclipse/hono/service/tenant/BaseTenantService.java

```diff
@@ -0,0 +1,345 @@
+                payload = getRequestPayload(body);
+                log.debug("creating tenant [{}]", tenantId);
+                add(tenantId, payload, result -> reply(tenantMsg, result));
+                break;
+            case ACTION_UPDATE:
+                payload = getRequestPayload(body);
+                log.debug("updating tenant [{}]", tenantId);
+                update(tenantId, payload, result -> reply(tenantMsg, result));
+                break;
+            case ACTION_REMOVE:
+                log.debug("deleting tenant [{}]", tenantId);
+                remove(tenantId, result -> reply(tenantMsg, result));
+                break;
+            }
+        } catch (final ClassCastException e) {
+            log.debug("malformed request message [{}]", e.getMessage());
+            reply(tenantMsg, TenantResult.from(HTTP_BAD_REQUEST));
+        }
+    }
+
+    private void reply(final Message<JsonObject> request, final AsyncResult<TenantResult> asyncActionResult) {
+        if (asyncActionResult.succeeded()) {
+            if (asyncActionResult.result() != null) {
+                reply(request, asyncActionResult.result());
+            } else {
+                log.debug("result is null in reply to tenant request");
+                reply(request, TenantResult.from(HTTP_INTERNAL_ERROR));
+            }
+        } else {
+            request.fail(HTTP_INTERNAL_ERROR, "cannot process tenant management request");
+        }
+    }
+
+    /**
+     * Sends a response to a tenant request over the Vertx event bus.
+     *
+     * @param request The message to respond to.
+     * @param tenantResult The tenant result that should be conveyed in the response.
+     * @throws NullPointerException If request or tenantResult is null.
+     */
+    protected final void reply(final Message<JsonObject> request, final TenantResult tenantResult) {
+        final JsonObject body = request.body();
+        request.reply(TenantConstants.getServiceReplyAsJson(body.getString(TenantConstants.FIELD_TENANT_ID), tenantResult));
+    }
+
+    /**
+     * Check the request payload for validity.
+     *
+     * @param request The request that was sent via the vert.x event bus containing the payload to be checked.
+     * @return boolean The result of the check : {@link Boolean#TRUE} if the payload is valid, {@link Boolean#FALSE} otherwise.
+     * @throws NullPointerException If the request is null.
+     */
+    private boolean isValidRequestPayload(final JsonObject request) {
+        final JsonObject payload = request.getJsonObject(TenantConstants.FIELD_PAYLOAD, new JsonObject());
+        try {
+            final JsonArray adapters = payload.getJsonArray(TenantConstants.FIELD_ADAPTERS);
+            if (adapters != null) {
+                if (adapters.size() == 0) {
+                    return false;
+                }
+                if (adapters.stream().filter(adapterDetails -> ((JsonObject) adapterDetails).getString(FIELD_ADAPTERS_TYPE) == null)
```

Comment:
can we agree to qualify references to constants that are not defined locally with the defining class? and consequently replace `import static TenantConstants.*` with `import TenantConstants`?

Reply:
Yes (you are so right). Will fix that.